server/core/runtime/policy/conftest_executor.go

```diff
@@ -0,0 +1,112 @@
+package policy
+
+import (
+	"context"
+	"fmt"
+	"github.com/runatlantis/atlantis/server/events/metrics"
+	contextInternal "github.com/runatlantis/atlantis/server/neptune/context"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/events/models"
+)
+
+type sourceResolver interface {
+	Resolve(policySet valid.PolicySet) (string, error)
+}
+
+type policyFilter interface {
+	Filter(ctx context.Context, installationToken int64, repo models.Repo, prNum int, failedPolicies []valid.PolicySet) ([]valid.PolicySet, error)
+}
+
+type exec interface {
+	CombinedOutput(args []string, envs map[string]string, workdir string) (string, error)
+}
+
+const (
+	conftestScope = "conftest.policies"
+	internalError = "internal server error"
+)
+
+// ConfTestExecutor runs a versioned conftest binary with the args built from the project context.
+// Project context defines whether conftest runs a local policy set or runs a test on a remote policy set.
+type ConfTestExecutor struct {
+	SourceResolver sourceResolver
+	Exec           exec
+	PolicyFilter   policyFilter
+}
+
+// Run performs conftest policy tests against changes and fails if any policy does not pass. It also runs an all-or-nothing
+// filter that will filter out all policy failures based on the filter criteria.
+func (c *ConfTestExecutor) Run(prjCtx command.ProjectContext, executablePath, workdir string, envs map[string]string, extraArgs []string) (string, error) {
```

Comment:
The legacy code [passes](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/events/project_command_runner.go#L234) both the `ctx` and its `prjctx` wrapper across all of the step runners (plan, policy, apply, etc.). Rather than go through and dedup all of that, I want to keep this PR small and just use the `prjctx` wrapper here.